projects/04_cube_xr/CubeXr.cpp

```diff
@@ -201,17 +201,27 @@ void CubeXrApp::Setup()
     mScissorRect = {0, 0, GetWindowWidth(), GetWindowHeight()};
 }
 
+void CubeXrApp::DispatchRender()
```

Comment:
Maybe this belongs to `XR[Aware]Application` or something.
i.e. CubeXrApp -> XRApplication -> Application

Reply:
Perhaps, and this PR is definitely no the end result of what XR should look like, but maybe just the next step.
We still need to figure out what that proper "end" state looks like. I think if we have `CubeXrApp -> XRApplication -> Application` we would want XRApplication to support both XR and non-XR. There are also different ways XR can be handled by application (just doing what it does now with full submits per view, recording one command buffer for everything, multiview), so I think this per view logic belongs in the derived application level.
I am kinda thinking maybe we should add another class (like Renderer), where application just calls renderer->Render(), and there can be specific classes like XrRenderer, XrMultiviewRenderer, etc, then we can also use that renderer in a different application to render to a specific composition layer